pennylane/circuit_graph.py

```diff
@@ -12,307 +12,284 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-CircuitGraph
-============
+Circuit graph
+=============
 
 **Module name:** :mod:`pennylane.circuit_graph`
 
 .. currentmodule:: pennylane.circuit_graph
 
 This module contains the CircuitGraph class which is used to generate a DAG (directed acyclic graph)
-based on operation and observable queues.
+representation of a quantum circuit.
 
-.. raw:: html
 
-    <h3>Summary</h3>
+Classes
+-------
 
 .. autosummary::
-    CircuitGraph
+   CircuitGraph
+   Layer
+   LayerData
 
-.. raw:: html
 
-    <h3>Code details</h3>
+CircuitGraph methods
+--------------------
+
+.. currentmodule:: pennylane.circuit_graph.CircuitGraph
+
+.. autosummary::
+   operations
+   observables
+   graph
+   wire_indices
+   ancestors
+   descendants
+   ancestors_in_order
+   descendants_in_order
+   _in_topological_order
+   layers
+   iterate_layers
+   update_node
+
+.. currentmodule:: pennylane.circuit_graph
+
+Code details
+~~~~~~~~~~~~
 """
 from collections import namedtuple
 import networkx as nx
 
 from .utils import _flatten
 
 
-Command = namedtuple("Command", ["name", "op", "return_type", "idx"])
-Layer = namedtuple("Layer", ["op_idx", "param_idx"])
+def _by_idx(x):
+    """Sorting key for Operations: queue index aka temporal order."""
+    return x.queue_idx
+
+
+Layer = namedtuple("Layer", ["ops", "param_inds"])
+"""Parametrized layer of the circuit.
+
+Args:
+
+    ops (list[Operation]): parametrized operations in the layer
+    param_inds (tuple[int]): corresponding free parameter indices
+"""
+#TODO define what a layer is
+
+LayerData = namedtuple("LayerData", ["pre_ops", "ops", "param_inds", "post_ops"])
```

Comment:
is `param_inds` short for indices? Perhaps we should stick with the `idxs` abbreviation, or use the full word instead.